lib/ukallocbbuddy/bbuddy.c

```diff
@@ -81,6 +84,8 @@ struct uk_bbpalloc {
 	chunk_head_t *free_head[FREELIST_SIZE];
 	chunk_head_t free_tail[FREELIST_SIZE];
 	struct uk_bbpalloc_memr *memr_head;
+	uk_spinlock level_locks[FREELIST_SIZE];
```

Comment:
Could add a comment for the "uk_bbpalloc" struct and can include a brief description for the level / global lock(s) usage (for example why they are needed) e.g. can be parts from the PR "Description of changes" section.